32-bit support

lfittl opened the pull request:
Fixes #231

lfittl committed:
Optionally support 32-bit builds

This can be useful when building for WASM targets, as well as for legacy
systems that run on 32-bit.

The relevant code is enabled at compile time by checking the pointer
size (__SIZEOF_POINTER__ == 4).

lfittl committed:
Correctly handle AclMode as an uint64 (not "long")

This mainly helps with resolving compiler warnings, and should not have
any effective difference, since the relevant struct (RTEPermissionInfo)
is not actually used in raw parsing.

lfittl committed:
examples/scan: Use correct format specifier for n_tokens

lfittl commented on protobuf/pg_query.proto:
```diff
@@ -1272,7 +1272,7 @@ message RTEPermissionInfo
   uint32 relid = 1 [json_name="relid"];
   bool inh = 2 [json_name="inh"];
-  int64 required_perms = 3 [json_name="requiredPerms"];
+  uint64 required_perms = 3 [json_name="requiredPerms"];
```
As noted in the commit message, this is okay to do only because this struct isn't actually used in raw parse trees. Otherwise this would be an API break (which may be okay in the context of how we use these protobufs, but at least we should talk about doing this outside of major releases).

lelit commented:
FYI, I tried out this PR and it fixes the problem, all [CI tests](https://github.com/lelit/pglast/actions/runs/7363134571/job/20042296417#step:5:1136) are green again!
Thank you!

lfittl committed:
Keep pg_config overrides in separate pg_config_overrides.h file

We've previously managed these in the Makefile inline, but that is
very hard to read, and not necessary, since we can simply concatenate
the override file to pg_config.h during the source extraction process.

anuraaga reviewed (commented):
It's a bit tricky to verify this with go-pgquery since I use the C sources from pg_query_go instead of these ones to make sure they align, but I think this should fix my issue well. Thanks a lot!

anuraaga commented:
Just to close the loop, after updating I could remove my 32-bit support patch. Thanks!

https://github.com/wasilibs/go-pgquery/pull/22/files#diff-4d79e217dbbb7c7f611bfd2e2a6111d14e17bff3f3dbafac24d6314e5eeafc03L58
